eth/filters/api.go

```diff
@@ -218,15 +219,25 @@ func (s *PublicFilterAPI) newLogFilter(earliest, latest int64, addresses []commo
 	filter.SetEndBlock(latest)
 	filter.SetAddresses(addresses)
 	filter.SetTopics(topics)
-	filter.LogCallback = func(log *vm.Log, removed bool) {
+
+	emitLog := func(log *vm.Log, removed bool) {
```

Comment:
Could we create function that returns a lambda rather than creating two separate functions here? 

```go
func getEmitter(...) {
    if emitting {
        return func(...) {
        }
    }
    return func(...) { }
}
```

Or move the logic in the lambda itself maybe and create one rather than two

```go
emitter := func(emit chan<- *event.Event) func(log *vm.Log, removed bool) {
    return func(log *vm.Log, removed bool) {
        if emit != nil {
            // logic
        } else {
            // logic
        }
    }
}
```